app/controllers/api/user_blocks_controller.rb

```diff
@@ -12,5 +15,33 @@ def show
     rescue ActiveRecord::RecordNotFound
       raise OSM::APINotFoundError
     end
+
+    def create
+      raise OSM::APIBadUserInput, "No user was given" unless params[:user]
+
+      user = User.visible.find_by(:id => params[:user])
+      raise OSM::APINotFoundError unless user
+      raise OSM::APIBadUserInput, "No reason was given" unless params[:reason]
+      raise OSM::APIBadUserInput, "No period was given" unless params[:period]
+
+      period = Integer(params[:period], :exception => false)
+      raise OSM::APIBadUserInput, "Period should be a number of hours" unless period
```

Comment:
I'm wondering if the permitted values should be more in line with user_block_periods in settings.yml:

```
# Periods (in hours) which are allowed for user blocks
user_block_periods: [0, 1, 3, 6, 12, 24, 48, 96, 168, 336, 731, 4383, 8766, 87660]
```

Are values not explicitly mentioned there ok for the UI, or would they cause issues in dropdown lists?

Reply:
Block durations are not stored directly. Dropdowns show the closes value to the remaining time: https://github.com/openstreetmap/openstreetmap-website/pull/5011